src/main/kotlin/frc/robot/subsystems/hood/HoodIOReal.kt

```diff
@@ -0,0 +1,58 @@
+package frc.robot.subsystems.hood
+
+import com.ctre.phoenix.motorcontrol.FeedbackDevice
+import com.ctre.phoenix.motorcontrol.can.TalonSRX
+import com.ctre.phoenix6.controls.PositionTorqueCurrentFOC
+import com.ctre.phoenix6.hardware.TalonFX
+import edu.wpi.first.math.geometry.Rotation2d
+import edu.wpi.first.units.Angle
+import edu.wpi.first.units.MutableMeasure
+import edu.wpi.first.units.Units
+import frc.robot.lib.Utils
+import kotlin.math.sign
+
+class HoodIOReal : HoodIO {
+    override val inputs = LoggedHoodInputs()
+    private val motor: TalonFX = TalonFX(-1) // TODO: Replace with real motor port
+    private val encoder = TalonSRX(-1) // TODO: Replace with real Port
+
+    private val positionControl = PositionTorqueCurrentFOC(0.0)
+
+    init {
+        motor.configurator.apply(HoodConstants.MOTOR_CONFIGURATION)
+
+        encoder.configSelectedFeedbackSensor(FeedbackDevice.CTRE_MagEncoder_Absolute)
+    }
+
+    private fun getEncoderPosition(): Double {
+        val encoderTicksPerRevolution = HoodConstants.ENCODER_TICKS_PER_REVOLUTION.toDouble()
+        val encoderPosition = encoder.selectedSensorPosition % encoderTicksPerRevolution
+        val normalizedPosition = encoderPosition / encoderTicksPerRevolution - HoodConstants.ABSOLUTE_ENCODER_OFFSET.get()
+        return Utils.normalize(Rotation2d.fromRotations(normalizedPosition)).rotations
+    }
+
+    override fun updateInternalEncoder() {
+        motor.setPosition(getEncoderPosition())
+    }
+
+    override fun setAngle(angle: MutableMeasure<Angle>) {
+        val error = angle.minus(inputs.absoluteEncoderAngle)
+        motor.setControl(
+            positionControl
+                .withPosition(inputs.internalAngle.plus(error).`in`(Units.Rotations))
```

Comment:
The update of the internal encoder is delayed. Updating the internal encoder is necessary for gravity calculations. If we switch to a CANCoder this can all be simplified of course.